VotingApplication/VotingApplication.Web.Api.Tests/E2E/BasicPollTests.cs

```diff
@@ -0,0 +1,603 @@
+
+                foreach (Option option in _defaultBasicPoll.Options)
+                {
+                    string truncatedDescription = option.Description
+                                                          .Split(' ')
+                                                          .Aggregate((prev, curr) => (curr.Length + prev.Length >= truncatedTextLimit) ?
+                                                                                     prev : prev + " " + curr);
+
+                    if (truncatedDescription != option.Description)
+                    {
+                        truncatedDescription += "... Show More";
+                    }
+                    expectedDescriptions.Add(truncatedDescription);
+                }
+
+                CollectionAssert.AreEquivalent(expectedDescriptions,
+                                               optionDescriptions.Select(o => o.Text).ToList());
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_VotingOnOption_NavigatesToResultsPage()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _defaultBasicPoll.UUID);
+                IReadOnlyCollection<IWebElement> buttons = _driver.FindElements(By.TagName("Button"));
+                buttons.First(b => b.Text == "Vote").Click();
+
+                VoteClearer voterClearer = new VoteClearer(_context);
+                voterClearer.ClearLast();
+
+                Assert.IsTrue(_driver.Url.StartsWith(SiteBaseUri + "Poll/#/Results/" + _defaultBasicPoll.UUID));
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_DefaultPoll_ShowsResultsButton()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _defaultBasicPoll.UUID);
+                IReadOnlyCollection<IWebElement> anchors = _driver.FindElements(By.TagName("a"));
+                IWebElement resultsLink = anchors.FirstOrDefault(a => a.Text == "Go to results");
+
+                Assert.IsTrue(resultsLink.IsVisible());
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_AfterVoting_VoteIsRemembered()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _defaultBasicPoll.UUID);
+
+                IReadOnlyCollection<IWebElement> buttons = _driver.FindElements(By.TagName("Button"));
+                buttons.First(b => b.Text == "Vote").Click();
+
+                _driver.Navigate().GoToUrl(_driver.Url.Replace("Results", "Vote"));
+
+                IWebElement selectedOption = _driver.FindElements(By.CssSelector(".selected-option")).Single();
+
+                VoteClearer voterClearer = new VoteClearer(_context);
+                voterClearer.ClearLast();
+
+                Assert.IsTrue(selectedOption.IsVisible());
+            }
+        }
+        #endregion
+
+        #region Invite Only Config
```

Comment:
Really not a fan of regions in general. What benefit is it giving us here? Visual Studio lets you collapse class definitions down; adding these regions just lets you collapse one line down to one line?